spec/components/email/sender_spec.rb

```diff
@@ -351,6 +351,34 @@
     end
   end
 
+  context "with attachments" do
+    let(:upload) do
+      UploadCreator.new(
+        file_from_fixtures("small.pdf", "pdf"),
+        "small.pdf"
+      ).create_for(Discourse.system_user.id)
+    end
+    let(:post) { Fabricate(:post, raw: "**hello** <a class=\"attachment\" href=\"#{upload.url}\">small.pdf</a>") }
+
+    before do
+      SiteSetting.authorized_extensions = 'pdf'
+    end
+
+    it "adds attachments to the email" do
+      post.link_post_uploads
+
+      message = Mail::Message.new to: 'disc@ourse.org', body: 'some content'
+      message.header['X-Discourse-Post-Id']  = post.id
+      message.header['X-Discourse-Topic-Id'] = post.topic_id
+      message.stubs(:deliver_now)
```

Comment:
We're tying to avoid the use of `stubs` in our code base moving forward. Do have a read through https://guides.rubyonrails.org/testing.html#testing-your-mailers to see the helpers that Rails provide.